include/fmt/core.h

```diff
@@ -933,7 +959,18 @@ template <typename T> class buffer {
   }
 
   /** Appends data to the end of the buffer. */
-  template <typename U> void append(const U* begin, const U* end);
+  template <typename U>
+  FMT_CONSTEXPR void append(const U* begin, const U* end) {
```

Comment:
The definition of `append` (along with `make_checked`) is moved from `format.h` because constexpr does not apply to external linkage.